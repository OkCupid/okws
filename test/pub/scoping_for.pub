{%
// The body of for loops have their scope.
 locals {x = "top_of_file\n"};
 for (whatever, [null]) {
     locals { x: "in_for_loop\n"};
     print(x); // <-------------------- in_for_loop
 }

 print(x); // <---------------- top_of_file

 // The empty clause of for loops does not push its own scope.
 for (whatever, []) {
 } empty {
     locals { x: "empty_clause\n"};
     print (x); // <--------------------- empty_clause
 }
 print(x); // <------------------- empty_clause
%}
