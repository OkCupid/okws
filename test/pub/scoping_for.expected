in_for_loop
top_of_file
empty_clause
empty_clause

